### PR TITLE
[LM-12386] Added custom widget support in chat bubble and conversation list

### PR DESCRIPTION
**Pull request by TyagiLikeMinds**
## Description

**Features**
- Added custom widget support in chat
**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [ ] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/MessageList/LMChatMessageListViewModel.swift [hunk 1]
```diff
@@ -1591,6 +1600,10 @@ extension LMChatMessageListViewModel {
                 .ogTags(currentDetectedOgTags)
             currentDetectedOgTags = nil
         }
+        var metadata: [String : Any] = [:]
+        metadata["chat_data"] = "data"
```
Comment (ishaan1607) [inline LMChatCore_iOS/LMChatCore_iOS/Source/Module/MessageList/LMChatMessageListViewModel.swift]: What this? Why is this required?

File: LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMChatMessageListView.swift [hunk 2]
```diff
@@ -94,8 +94,9 @@ open class LMChatMessageListView: LMView {
...
+            public var metadata: [String:Any]?
             
-            public init(messageId: String, memberTitle: String?, memberState: Int?, message: String?, timestamp: Int?, reactions: [Reaction]?, attachments: [Attachment]?, replied: [Message]?, isDeleted: Bool?, createdBy: String?, createdByImageUrl: String?, createdById: String?, isIncoming: Bool?, messageType: Int, createdTime: String?, ogTags: OgTags?, isEdited: Bool?, attachmentUploaded: Bool?, isShowMore: Bool, messageStatus: LMMessageStatus?, tempId: String?, hideLeftProfileImage: Bool?, pollData: LMChatPollView.ContentModel?) {
+            public init(messageId: String, memberTitle: String?, memberState: Int?, message: String?, timestamp: Int?, reactions: [Reaction]?, attachments: [Attachment]?, replied: [Message]?, isDeleted: Bool?, createdBy: String?, createdByImageUrl: String?, createdById: String?, isIncoming: Bool?, messageType: Int, createdTime: String?, ogTags: OgTags?, isEdited: Bool?, attachmentUploaded: Bool?, isShowMore: Bool, messageStatus: LMMessageStatus?, tempId: String?, hideLeftProfileImage: Bool?, pollData: LMChatPollView.ContentModel?, metadata: [String:Any]?) {
```
Comment (ishaan1607) [inline LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMChatMessageListView.swift:99]: add new line for params, it is not readable

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMMessageCell/LMChatCustomCell.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        public let message: LMChatMessageListView.ContentModel.Message?
+        public var isSelected: Bool = false
+    }
+    
+    public var data: ContentModel?
+    public var index: IndexPath?
+    
+    public let customMessageLabel: UILabel = {
+        let label = UILabel()
+        label.text = "This is a custom view cell, modify it using LMUIComponent to generate your custom view"
```
Comment (ishaan1607) [inline LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMMessageCell/LMChatCustomCell.swift]: why is this required?

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMMessageCell/LMChatCustomCell.swift (hunk 3 above)
Comment (TyagiLikeMinds) [reply LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMMessageCell/LMChatCustomCell.swift]: Fixed

File: LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMChatMessageListView.swift (hunk 2 above)
Comment (TyagiLikeMinds) [reply LMChatUI_iOS/LMChatUI_iOS/Source/UI/MessageList/LMChatMessageListView.swift:99]: Fixed

File: LMChatCore_iOS/LMChatCore_iOS/Source/Module/MessageList/LMChatMessageListViewModel.swift (hunk 1 above)
Comment (TyagiLikeMinds) [reply LMChatCore_iOS/LMChatCore_iOS/Source/Module/MessageList/LMChatMessageListViewModel.swift]: Fixed

**Review by ishaan1607 — APPROVED** [review]
